prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -278,6 +278,21 @@ void rxr_pkt_handle_readrsp_sent(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_en
 	}
 }
 
+void rxr_pkt_handle_readrsp_send_completion(struct rxr_ep *ep,
+					    struct rxr_pkt_entry *pkt_entry)
+{
+	struct rxr_tx_entry *tx_entry;
+	struct rxr_readrsp_hdr *readrsp_hdr;
+
+	readrsp_hdr = (struct rxr_readrsp_hdr *)pkt_entry->pkt;
+
+	tx_entry = (struct rxr_tx_entry *)pkt_entry->x_entry;
```

Comment:
Can you re-add the assert for cq_entry.flags & FI_READ? Or add some sort of assert to validate that this tx_entry is from the readrsp pool?

Reply:
Thank you, added back in new revision.